rpc/nodeagent/security/keyvault/key/moc_nodeagent_key.proto

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache v2.0 license.
+
+syntax = "proto3";
+option go_package = "github.com/microsoft/moc/rpc/nodeagent/security";
+package moc.nodeagent.security;
+
+import "google/protobuf/wrappers.proto";
+import "moc_common_common.proto";
+import "moc_common_security.proto";
+import "google/protobuf/timestamp.proto";
+
+
+message KeyRequest {
+	repeated Key Keys = 1;
+	Operation OperationType = 2;
+}
+
+message KeyResponse {
+	repeated Key Keys = 1;
+	google.protobuf.BoolValue Result = 2;
+	string  Error = 3;
+}
+
+message KeyOperationRequest {
+	Key key = 1;
+	string Data = 2 [(sensitive) = true];
+	Algorithm algorithm = 3;
+	KeyOperation OBSOLETE_OperationType = 4 [deprecated=true];
+	SignVerifyParams  SignVerifyParams = 5;
+	ProviderAccessOperation OperationType = 6;
+}
+
+message KeyOperationResponse {
+	string Data = 1 [(sensitive) = true];
+	google.protobuf.BoolValue Result = 2;
+	string  Error = 3;
+	Key key = 4;
+}
+
+message PrivateKeyWrappingInfo {
+	string WrappingKeyName = 1;
+	bytes WrappingKeyPublic = 2 [(sensitive) = true];
+	KeyWrappingAlgorithm WrappingAlgorithm = 3;
+}
+
+message SignVerifyParams {
+	JSONWebKeySignatureAlgorithm  algorithm  = 1;
+	string signature = 2  [(sensitive) = true];
+}
+
+message Key {
+	string name = 1;
+	string id = 2;
```

Comment:
Is we have a lot in common can we refactor the cloudagent and nodeagent Key data?

Reply:
is this resolved?